workspaces/liferay-sample-workspace/client-extensions/liferay-sample-etc-node/util/configMap.js

```diff
@@ -0,0 +1,26 @@
+import config from '../config.js';
+import fs from 'fs';
+import path from 'path';
+
+const configTreePath = '/etc/liferay/lxc';
+
+async function* walk(dir) {
+	if (fs.existsSync(dir) === false) return;
+	for await (const dirent of await fs.promises.opendir(dir)) {
```

Comment:
is this reading all the dotfiles as well or does this exclude them somehow?

Reply:
I am not for sure.

Reply:
I'm asking because the configmaps are all symlinks to files stored under a hidden directory. it does that for versioning so potentially you could also have older versions of the configmap also on the filesystem. 

if this logic walks through all directories including the ones like the following you could potentially be reading old config info: 

```
drwxr-xr-x 2 root root   60 May  2 18:48 ..2023_05_02_18_48_01.747363171
lrwxrwxrwx 1 root root   31 May  2 18:48 ..data -> ..2023_05_02_18_48_01.747363171
```

Reply:
yeah, you probably want to ignore any files (and/or skip descending into directories) that have name length >2 and start with double dot `..`

Reply:
yeah that's why in my example i just explicitly looked into a list of configmap directories rather than walking through everything

Reply:
but we should walk everything (minus the known ignorable paths) so that we don't miss new members of the contract as it evolves.

Reply:
you mean there will be new configmaps under /etc/liferay/lxc?

Reply:
I think he means that today we have two "metadata" config maps that we mount under /etc/liferay/lxc
/etc/liferay/lxc/dxp-metadata
/etc/liferay/lxc/ext-init-metadata

But next month or year we may add
/etc/liferay/lxc/ext-oidc-metadata
or something and we want this code to continue to work.